indexing-service/src/main/java/org/apache/druid/indexing/common/config/TaskConfig.java

```diff
@@ -184,14 +228,48 @@ public String getBaseDir()
   }
 
   @JsonProperty
-  public File getBaseTaskDir()
+  public List<StorageLocationConfig> getBaseTaskDirLocations()
+  {
+    return baseTaskDirLocations;
+  }
+
+  public List<StorageLocationConfig> getBaseTaskDirChildrenLocations(String child)
+  {
+    return getBaseTaskDirLocations().stream()
+                                    .map(location -> new StorageLocationConfig(
+                                        getChildFileFromLocation(location, child),
+                                        location.getMaxSize(),
+                                        location.getFreeSpacePercent()))
+                                    .collect(Collectors.toList());
+  }
+
+  private File getChildFileFromLocation(StorageLocationConfig location, String child)
+  {
+    return new File(location.getPath(), child);
+  }
+
+  private StorageLocationConfig getBaseTaskDirLocationForId(String taskId)
+  {
+    if (taskId == null) {
+      return null;
+    }
+    // TODO: consider the size of each of the base locations for distribution
+    // use uniformly RANDOM assignment
+    int index = taskId.hashCode() % baseTaskDirLocations.size();
```

Comment:
```suggestion
    int index = Math.abs(taskId.hashCode() % baseTaskDirLocations.size());
```

I think this can be used to eliminate following if statement checking, and is more easy to understand.

Reply:
I agree, however the bug analyzer complains that Math.abs(Integer.MIN_VALUE) is invalid

Reply:
The argument will never be Integer.MIN_VALUE logically but I had to do it in a roundabout manner to get past it.
Is there a way to suppress it?

Reply:
> I agree, however the bug analyzer complains that Math.abs(Integer.MIN_VALUE) is invalid

Which bug analyzer? spotbugs?

Reply:
Yes